src/mame/drivers/roland_pr100.cpp

```diff
@@ -0,0 +1,65 @@
+// license:BSD-3-Clause
+// copyright-holders:Valley Bell
+/****************************************************************************
+
+    Skeleton driver for Roland PR-100 Digital Sequencer.
+
+****************************************************************************/
+
+#include "emu.h"
+#include "cpu/z180/z180.h"
+
+class roland_pr100_state : public driver_device
+{
+public:
+	roland_pr100_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag)
+		, m_mpu(*this, "maincpu")
+	{
+	}
+
+	void pr100(machine_config &config);
+
+private:
+	void mem_map(address_map &map);
+	void io_map(address_map &map);
+
+	required_device<hd64180rp_device> m_mpu;
+};
+
+void roland_pr100_state::mem_map(address_map &map)
+{
+	map(0x0000, 0x9fff).rom().region("mpurom", 0);
+	map(0xb000, 0xefff).ram();	// stack pointer is initialized to 0xF000, the routine at 0x0325 writes to 0xB003
+	// But the routine at 0x02AD also writes to all bytes from 0x8000 to 0xFFFF (does "OUT0 ($38), ## select some RAM bank?)
+	// TODO: map(0xf000, 0xffff) seems to be one of the ROM banks 0xA000/0xB000/0xC000/0xD000/0xE000/0xF000
```

Comment:
The Z180/HD64180 architecture includes a built-in MMU that maps the 64K logical space to a 1M physical space (actually only 512K in this case due to the DIP version not having a pin to spare for A19). It would help to understand how that works.

Reply:
Oh, that's nice to know. Unfortunately my knwoledge goes only until the Z80 (not Z180), so all assumptions I did are based on what I know about the Z80 architecture.

If you want to do any research on this, feel free to do so. I won't be doing any "real" work on this driver anytime soon. (I still have lots of things to do with the CM-32P.)